src/runtime_src/hip/api/hip_memory.cpp

```diff
@@ -3,190 +3,270 @@
 
 #include <string>
 #include "core/common/error.h"
+#include "core/common/unistd.h"
 #include "hip/config.h"
 #include "hip/core/device.h"
 #include "hip/core/memory.h"
 #include "hip/hip_runtime_api.h"
 
 namespace xrt::core::hip
 {
+  static std::shared_ptr<device>
+  get_current_device()
+  {
+    // TODO: get REAL current hip device
+    auto dev = device_cache.get(0);
+    if (dev == nullptr)
+    {
+      if (hipInit(0) != hipSuccess)
+      {
+        throw std::runtime_error("hipInit() failed!");
+      }
+      dev = device_cache.get(0);
+    }
+    return dev;
+  }
 
   // Allocate memory on the device.
   static void
-  hip_malloc(void **ptr, size_t size)
+  hip_malloc(void** ptr, size_t size)
   {
-    throw std::runtime_error("Not implemented");
+    assert(ptr);
+    assert(size > 0);
+
+    auto dev = get_current_device();
+    assert(dev);
+
+    auto hip_mem = std::make_shared<xrt::core::hip::memory>(dev, size);
+    auto dev_addr = hip_mem->get_addr(address_type::hip_address_type_device);
+    if (dev_addr != 0)
+    {
+      memory_database::instance().insert_addr(address_type::hip_address_type_device, reinterpret_cast<uint64_t>(dev_addr), size, hip_mem);
```

Comment:
Do we really want to differentiate host vs device at this level? Anyways, lets go with this for now. I think, all of this need to be taken care by memory_database itself.